### PR TITLE
Web fixes

### PR DESCRIPTION
**Pull request by Karinon**
This is a collection of several smaller fixes:

- fixed issue where the databrowser has shown an `undefined`-flavour when none was selected under certain circumstances
- replaced plain HTTPResponses by JSONResponses when the passed data was indeed a JSON
- added error message when react didn't load properly (prior to that: No info at all)
- added error message when there are issues with the user information (LDAP unreachable, Redis down) in one of the react-views (prior to that: infinite load animation)
- some smaller visual changes

### CODE REVIEW

**Comment by Karinon** [comment]
The failed CI-Checks are black-formatting issues and you have already fixed them in your own PR, therefore I will not touch it

File: plugins/views.py [hunk 1]
```diff
@@ -324,32 +325,28 @@ def list_dir(request):
     except UserNotFoundError:
         # This user has no access to the underlying system and therefore
         # must not be able to get a file listing
-        return HttpResponse(
-            json.dumps(
-                {
-                    "status": "You are not allowed to see the folder listing",
-                    "folders": [],
-                }
-            )
```
Comment (antarcticrainforest) [inline plugins/views.py:333]: Nothing to complain here, but I do think there's more of that stuff in solr (are we using this at all). and history.

**Comment by antarcticrainforest** [comment]
I can't really say much, most of the changes are beyond my knowledge. The only comment I have is about the `HTTPResponse` -> `JSONRespons`. There's more of them :)

File: plugins/views.py (hunk 1 above)
Comment (Karinon) [reply plugins/views.py:333]: I just checked the ones in history/views.py (there are no in solr.py). They look kind of...special..
They use this `json.dumps` syntax but do not put json into it but simple strings...I think I will tackle them in a separate PR as I am a bit afraid of breaking something

**Pull request merged**